### DISCUSSION TITLE
Stochastic - error in SamplerReceiver

### DISCUSSION THREAD

**Discussion by MatiasAllay** [idaholab/moose#18046] [Q&A Modules: General] [answered]
Hi MOOSE,
I'm using the stochastic module to perform a SOBOL sensitivity analysis. However, I get the following error

`*** ERROR ***
runner4: The following error occurred in the object "stochastic", of type "SamplerReceiver".
runner4: 
runner4: The desired parameter 'Materials/velocity_k/value' was not located for the 'stochastic' object, it either does not exist or has not been declared as controllable`

Not sure what is going on... I get the same error for every parameter. Any idea?
Cheers,
Matias

**Comment by MatiasAllay** [comment, marked as answer]
Hi MOOSE,
My questions was not really precise, I leave here a simplified version of my problem. I would appreciate if someone can take a look. 
I am sure is some stupid mistake!

https://drive.google.com/drive/folders/1Fu-Yn80egK4Bep-PHbB5dE3G85_rKFWP?usp=sharing
Cheers,
Matias

**Reply by jiangwen84** [reply]
1. The first error message says that `porosity` is not declared as controllable. To make the control system work, the parameters need to be declared as controllable. See https://mooseframework.inl.gov/syntax/Controls/index.html 
2. After looking at `PorousFlowPorosityConst`, the `porosity` is a variable name and it probably cannot be declared as controllable variable. 

For your case, I recommend you use `MultiAppCommandLineControl. See https://mooseframework.inl.gov/source/controls/MultiAppCommandLineControl.html 

It is a more flexible way to set the parameters because parameter values are directly set through command line which does not require them to be controllable.

**Reply by MatiasAllay** [reply]
Hi,
Thanks for you reply. I took some time to perform some tests. The `MultiAppCommandLineControl` sounds like a excellent solution. However, when I apply it to my problem I get the error 

`execute_on: The sampler object, 'sobol', is being used by the 'cmdline' object, thus the 'execute_on' of the sampler must include 'PRE_MULTIAPP_SETUP'.`

But apparently, the Sampler/Sobol does not have the flag `execute_on` (https://mooseframework.inl.gov/moose/source/samplers/SobolSampler.html). Any idea how can I solve this?

Thanks a lot!
Matias

**Reply by aeslaughter** [reply]
Comment out the following line, recompile, and run with execute_on set as recommend. I am not remembering why that was disabled, but we will see if that gets it working for you.

 https://github.com/idaholab/moose/blob/48fe2f2d002b8e55afcb004bdc55bf124cafc4e7/modules/stochastic_tools/src/samplers/SobolSampler.C#L24

**Reply by MatiasAllay** [reply]
Thanks! That worked perfectly.
Matias

**Reply by aeslaughter** [reply]
Great, I will have to change that in MOOSE.
